src/argilla/client/feedback/training/schemas/base.py

```diff
@@ -39,43 +53,45 @@
 )
 from argilla.client.models import Framework
 from argilla.utils.dependency import require_dependencies, requires_dependencies
+from pydantic import BaseModel
 
 _LOGGER = logging.getLogger(__name__)
 
 if TYPE_CHECKING:
     import datasets
     import spacy
+    from argilla.client.feedback.dataset import FeedbackDataset
     from sentence_transformers import InputExample
 
-    from argilla.client.feedback.dataset import FeedbackDataset
 
+class TrainingData(ABC):
+    formatting_func: Optional[BaseModel] = None
+    defaults: Optional[BaseModel] = None
+    _formatting_func_return_types = None
+    _supported_frameworks_names: list = []
+
+    @property
+    def formatting_func_return_types(self) -> BaseModel:
+        return self._formatting_func_return_types
 
-TASK_STRUCTURE = {
-    "text_classification": {
-        "field": (TextField),
-        "question": (
-            LabelQuestion,
-            MultiLabelQuestion,
-            RatingQuestion,
-            RankingQuestion,
-        ),
-        "unification": (
-            LabelQuestionUnification,
-            MultiLabelQuestionUnification,
-            RatingQuestionUnification,
-            RankingQuestionUnification,
-        ),
-    },
-    "question_answering": {
-        "field": (TextField),
-        "question": (TextQuestion),
-        "unification": (),
-    },
-}
+    @property
+    def supported_frameworks(self) -> List[Framework]:
+        return [Framework(name) for name in self._supported_frameworks_names]
 
+    def __repr__(self) -> str:
+        return textwrap.dedent(
+            f"{self.__class__.__name__}" f"formatting_func:\n{self.formatting_func}, "
+            if self.formatting_func
+            else "" f"defaults:\n{self.defaults}), "
+            if self.defaults
+            else ""
+        )
```

Comment:
How do we expect this to render? I see the result a bit weird:
```python
textwrap.dedent(
    f"{'class_name'}" f"formatting_func:\n{'value'}, "
    if 'value'
    else "" f"defaults:\n{'defaults'}), "
    if 'defaults'
    else ""
)
# >>> 'class_nameformatting_func:\nvalue, '
```

Reply:
Thanks, this should be updated